### PR TITLE
expose with_public_ip var of metadata service

### PR DESCRIPTION
**Pull request by andreinechaev**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -141,3 +141,9 @@ variable "extra_ui_static_env_vars" {
   default     = {}
   description = "Additional environment variables for UI static app"
 }
+
+variable "with_public_ip" {
```
Comment (Erin-Boehmer) [inline variables.tf:145]: @andreinechaev if you expose, you'll need to add to the README as well

Comment (oavdeev) [reply variables.tf:145]: nit: now that I think of it we should probably make the description more descriptive too (I understand this PR copied it from the submodule, but still). Maybe something like "enable public IP assignment for the metadata service".

**Comment by oavdeev** [comment]
@Erin-Boehmer how did this work for you, since you added the option not to assign the IP in the first place in #37 ? 

I believe @andreinechaev ran into this issue with ECS not being able to pull images when it doesn't have IP address but runs in a public subnet. If it was a private subnet, I think it would be fine granted NAT gateway and Internet gateway are set up as usual.

We should maybe add a note that you should either
1. use a public subnet and remember to set `with_public_ip` to `true`
2. use a private subnet, then `with_public_ip` doesn't matter (?)
3. use a public subnet, leave `with_public_ip` to `false`, but configure the module to pull container images from a private repo accessible from within the VPC

I'm now trying to think what's the best way to communicate this to module users 🤔 Maybe add something in the description for the subnet setting too.

**Review by oavdeev — APPROVED** [review]
sorry for the delay, im going to include this into the next release

**Comment by leeyh20** [comment]
@oavdeev  Agree, i ran into the same issue today. 
I was running this:
https://github.com/outerbounds/terraform-aws-metaflow/blob/v0.7.1/examples/minimal/minimal_example.tf
And because ECS is set to automatically use a private ip, the ECS cannot pull the image for metadata service.

image

Therefore now the minimal example would not work, since the minimal example uses public subnets for the ECS.

According to AWS, private subnet should have `with_public_ip`  set to false.
Link: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_cannot_pull_image.html

**Pull request merged**